flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -137,6 +137,28 @@ class TableEnvironmentTest {
       .tableExists(ObjectPath.fromString(s"${tableEnv.getCurrentDatabase}.tbl1")))
   }
 
+  @Test
+  def testExecuteSqlWithCreateDropTemporaryTable(): Unit = {
```

Comment:
Testing only the simplest "create + drop" will be insufficient. At least we have to cover some illegal cases, adding some cases when temporary & normal table co-exists.

Reply:
BTW, will "show tables" command also show temporary tables?

Reply:
The ITCase `testTemporaryTableMaskPermanentTableWithSameName` test co-exists cases.

Reply:
`show tables` show both temporary and permanent tables

Reply:
I don't think `testTemporaryTableMaskPermanentTableWithSameName`  is enough either